src/mds/snap.cc

```diff
@@ -65,6 +65,15 @@ void SnapInfo::dump(Formatter *f) const
   f->close_section();
 }
 
+void SnapInfo::update_metadata(string& key, string& value) {
+  if (metadata.find(key) != metadata.end()) {
+    metadata.erase(key);
+    metadata.insert({key, value});
+  } else {
+    metadata.insert({key, value});
+  }
+}
```

Comment:
is it needed to erase the key first and then re-add the pair? How about using https://en.cppreference.com/w/cpp/container/map/insert_or_assign.html:
```suggestion
  metadata.insert_or_assign(key, value);
```